File: task3/main.swift
import Foundation

// Ввод количества чисел для проверки
print("Введите число: ", terminator: "")
guard let input = readLine(), let n = Int(input) else { exit(0) }

var count = 0

// Проверка каждого числа на одинаковые цифры
for _ in 0..<n {
    guard let numInput = readLine(), var num = Int(numInput) else { continue }
    let lastDigit = num % 10  // Берём последнюю цифру для сравнения
    num /= 10
    var sameDigit = true
    
    // Проверяем, все ли цифры совпадают с последней
    while num > 0 {
        let currentDigit = num % 10
        if currentDigit != lastDigit {
            sameDigit = false
            break
        }
        num /= 10
    }
    
    if sameDigit {
        count += 1  // Увеличиваем счётчик, если все цифры одинаковы
    }
}

print("Количество: \(count)")  // Вывод результата

File: task3/main.go
package main

import "fmt"

func main() {
	var n, num, count int
	fmt.Print("Введите число: ")
	fmt.Scan(&n)

	for i := 0; i < n; i++ {
		fmt.Scan(&num)
		lastDigit := num % 10
		num /= 10
		sameDigit := true

		for num > 0 {
			currentDigit := num % 10
			if currentDigit != lastDigit {
				sameDigit = false
				break
			}
			num /= 10
		}

		if sameDigit == true {
			count++
		}
	}

	fmt.Println("Количество: ", count)
}
